cmd/lakectl/cmd/fs.go

```diff
@@ -409,6 +413,159 @@ var fsRmCmd = &cobra.Command{
 	},
 }
 
+const (
+	fsDownloadCmdMinArgs = 1
+	fsDownloadCmdMaxArgs = 2
+
+	fsDownloadParallelDefault = 6
+)
+
+var fsDownloadCmd = &cobra.Command{
+	Use:   "download <path uri> [<destination path>]",
+	Short: "Download object(s) from a given repository path",
+	Args:  cobra.RangeArgs(fsDownloadCmdMinArgs, fsDownloadCmdMaxArgs),
+	Run: func(cmd *cobra.Command, args []string) {
+		pathURI := MustParsePathURI("path", args[0])
+		flagSet := cmd.Flags()
+		direct := MustBool(flagSet.GetBool("direct"))
+		recursive := MustBool(flagSet.GetBool("recursive"))
+		parallel := MustInt(flagSet.GetInt("parallel"))
+
+		if parallel < 1 {
```

Comment:
Do we not want to bound it from above as well?

Reply:
Prefer to limit just things that will not work and let the user abuse if needed. 
Checked the upload implementation - it is unbound (will need to fix it too) - you get a goroutine for each file. which means that if you do a recursive upload of a large file system, you will probably crash.